libnmstate/nm/connection.py

```diff
@@ -137,36 +137,83 @@ def commit_profile(connection_profile, save_to_disk=True, nmdev=None):
     con_profile.commit(save_to_disk, nmdev)
 
 
+class ConnectionSetting(object):
+
+    def __init__(self, con_setting=None):
+        self._setting = con_setting
+
+    def create(self, con_name, iface_name, iface_type):
+        con_setting = nmclient.NM.SettingConnection.new()
+        con_setting.props.id = con_name
+        con_setting.props.interface_name = iface_name
+        con_setting.props.uuid = str(uuid.uuid4())
+        con_setting.props.type = iface_type
+        con_setting.props.autoconnect = True
+        con_setting.props.autoconnect_slaves = (
+            nmclient.NM.SettingConnectionAutoconnectSlaves.YES)
+
+        self._setting = con_setting
+        self._logging_connection_info('create_setting')
+
+    def import_by_profile(self, con_profile):
+        base = con_profile.get_setting_connection()
+        new = nmclient.NM.SettingConnection.new()
+        new.props.id = base.props.id
+        new.props.interface_name = base.props.interface_name
+        new.props.uuid = base.props.uuid
+        new.props.type = base.props.type
+        new.props.autoconnect = True
```

Comment:
This can be removed if this is set in `__ini__` IMHO. It is also not nice to hardcode this in multiple places.

Reply:
It is expressing what we do today: Hard coding it, overwriting anything we touch to have it enabled.
When the need will arise we will change it.